File: queue.c
#include<stdio.h>

#define SIZE 5

int items[SIZE], front=-1, rear=-1;

// add element to the queue
void enQueue(int value){
    if(rear == SIZE-1){
        printf("Queue is full\n");
    }else{
        if(front == -1){
            front = 0;
        }
        rear++;
        items[rear] = value;
        printf("Inserted %d\n", value);
    }
}

// remove element from the queue
void deQueue(){
    if(front == -1){
        printf("Queue is empty\n");
    }else{
        front++;
        printf("Deleted %d\n", items[front]);
        if(front>rear){
            front=rear=-1;
        }
    }
}

// function to print the queue
void display(){
    if(rear == -1){
        printf("Queue is empty");
    }else{
        printf("Queue element are: ");
        for(int i=front; i<=rear; i++){
            printf("%d ", items[i]);
        }
    }
    printf("\n");
}

int main(){
    deQueue();

    enQueue(1);
    enQueue(2);
    enQueue(3);
    enQueue(4);
    enQueue(5);

    enQueue(6);

    deQueue();

    display();

    return 0;
}

File: Data Structures/Data Structure-l/queue.js
class Queue {
  constructor(size = 5) {
    this.items = new Array(size);
    this.front = -1;
    this.rear = -1;
    this.size = size;
  }

  /**
   * Adds an element to the queue
   * @param {number} value
   */
  enQueue(value) {
    if (this.rear === this.size - 1) {
      console.log("Queue is full");
    } else {
      if (this.front === -1) {
        this.front = 0;
      }
      this.rear++;
      this.items[this.rear] = value;
      console.log(`Inserted ${value}`);
    }
  }

  /**
   * Removes an element from the queue
   */
  deQueue() {
    if (this.front === -1 || this.front > this.rear) {
      console.log("Queue is empty");
    } else {
      console.log(`Deleted ${this.items[this.front]}`);
      this.front++;
      if (this.front > this.rear) {
        this.front = this.rear = -1;
      }
    }
  }

  /**
   * Displays the queue elements
   */
  display() {
    if (this.rear === -1) {
      console.log("Queue is empty");
    } else {
      console.log(
        "Queue elements are:",
        this.items.slice(this.front, this.rear + 1).join(" ")
      );
    }
  }
}

/**
 * Example usage
 */
const queue = new Queue();
queue.deQueue();

queue.enQueue(1);
queue.enQueue(2);
queue.enQueue(3);
queue.enQueue(4);
queue.enQueue(5);

queue.enQueue(6);

queue.deQueue();
queue.deQueue();

queue.display();
